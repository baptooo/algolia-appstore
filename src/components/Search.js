import React  from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
  container: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  textField: {
    margin: '25px 0',
    width: '100%',
  },
  menu: {
    width: 200,
  },
});

const Search = ({ onChange, classes, query }) => (
  <div style={{ textAlign: 'center' }}>
    <TextField
      value={query}
      className={classes.textField}
      name="term"
      label="Search for an app by name"
      onChange={(evt) => onChange(evt.target.value)}
      inputProps={{
        autoComplete: 'off'
      }}
    />
  </div>
);

Search.propTypes = {
  onChange: PropTypes.func.isRequired
};

export default withStyles(styles)(Search);