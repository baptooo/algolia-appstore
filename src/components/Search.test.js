import React from 'react';
import renderer from 'react-test-renderer';
import Search from './Search';

const noop = () => {};

describe('(Component) Search', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Search onChange={noop}/>).toJSON();

    expect(tree).toMatchSnapshot();
  })
});