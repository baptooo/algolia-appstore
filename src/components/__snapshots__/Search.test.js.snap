// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Search renders correctly 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-4 Search-textField-2"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-13 MuiInputLabel-root-8 MuiInputLabel-formControl-9 MuiInputLabel-animated-12"
      data-shrink={false}
      htmlFor={undefined}
    >
      Search for an app by name
    </label>
    <div
      className="MuiInput-root-18 MuiInput-formControl-19 MuiInput-underline-22"
    >
      <input
        aria-invalid={false}
        aria-required={false}
        autoComplete="off"
        autoFocus={undefined}
        className="MuiInput-input-26"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="term"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
</div>
`;
